src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_modal.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  EuiCard,
+  EuiModal,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  useGeneratedHtmlId,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiRadio,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import type { WorkflowDetailDto, WorkflowListItemDto } from '@kbn/workflows';
+import { capitalize } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { MANUAL_TRIGGERS_DESCRIPTIONS } from '../../../../common/translations';
+import { WorkflowExecuteEventForm } from './workflow_execute_event_form';
+import { WorkflowExecuteManualForm } from './workflow_execute_manual_form';
+
+type TriggerType = 'manual' | 'alert' | 'scheduled';
+
+export function WorkflowExecuteModal({
+  workflow,
+  onClose,
+  onSubmit,
+}: {
+  workflow: WorkflowDetailDto | WorkflowListItemDto;
+  onClose: () => void;
+  onSubmit: (data: Record<string, any>) => void;
+}) {
+  const modalTitleId = useGeneratedHtmlId();
+  const enabledTriggers =
+    workflow?.definition.triggers.filter((t: any) => t.enabled).map((t: any) => t.type) || [];
+
+  const [selectedTrigger, setSelectedTrigger] = useState<TriggerType>(enabledTriggers[0]);
+
+  const [executionInput, setExecutionInput] = useState<string>('');
+  const [executionInputErrors, setExecutionInputErrors] = useState<string | null>(null);
+
+  const handleSubmit = () => {
+    onSubmit(JSON.parse(executionInput));
+    onClose();
+  };
+
+  const handleChangeTrigger = (trigger: TriggerType): void => {
+    setExecutionInput('');
+    setSelectedTrigger(trigger);
+  };
+
+  return (
+    <EuiModal
+      aria-labelledby={modalTitleId}
+      onClose={onClose}
+      maxWidth={1400}
+      style={{ width: '1200px', height: '100vh' }}
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle id={modalTitleId}>Run Workflow</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiFlexGroup direction="row" gutterSize="l">
+          {enabledTriggers.map((trigger) => (
```

Comment:
It would be probably better to use styled radio group for semantic, like here https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/fleet_extensions/asset_boxed_radio_group.tsx

<img width="1866" height="900" alt="Image" src="https://github.com/user-attachments/assets/3c2db6f1-c39e-4240-bc5a-4f1f20b9c30a" />